internal/storage/v2/memory/memory.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package memory
+
+import (
+	"context"
+	"sync"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	conventions "go.opentelemetry.io/collector/semconv/v1.16.0"
+
+	v1 "github.com/jaegertracing/jaeger/internal/storage/v1/memory"
+	v2api "github.com/jaegertracing/jaeger/internal/storage/v2/api/tracestore"
+	"github.com/jaegertracing/jaeger/internal/tenancy"
+)
+
+const noServiceName = "OTLPResourceNoServiceName"
+
+// Store is an in-memory store of traces
+type Store struct {
+	sync.RWMutex
+	// Each tenant gets a copy of default config.
+	// In the future this can be extended to contain per-tenant configuration.
+	defaultConfig v1.Configuration
+	perTenant     map[string]*Tenant
+}
+
+// Tenant is an in-memory store of traces for a single tenant
+type Tenant struct {
+	sync.RWMutex
+	ids        []pcommon.TraceID
+	traces     map[pcommon.TraceID]ptrace.Traces
+	services   map[string]struct{}
+	operations map[string]map[v2api.Operation]struct{}
+	config     v1.Configuration
+	index      int
+}
+
+// NewStore creates an unbounded in-memory store
+func NewStore() *Store {
+	return WithConfiguration(v1.Configuration{MaxTraces: 0})
+}
+
+// WithConfiguration creates a new in memory storage based on the given configuration
+func WithConfiguration(cfg v1.Configuration) *Store {
+	return &Store{
+		defaultConfig: cfg,
+		perTenant:     make(map[string]*Tenant),
+	}
+}
+
+func newTenant(cfg v1.Configuration) *Tenant {
```

Comment:
there's no reason to keep copying configuration for each tenant, take it as pointer